json_serialization_writer.go

```diff
@@ -779,6 +787,7 @@ func (w *JsonSerializationWriter) WriteAdditionalData(value map[string]interface
 
 // Close clears the internal buffer.
 func (w *JsonSerializationWriter) Close() error {
```

Comment:
Assuming that most users aren't reusing writers, you could be more efficient here by putting the buffer back in a pool and reusing it on instantiation. However, that would also prohibit reusing the writer. You could add `Reset` for the reuse case, which is the pattern that the stdlib uses.